spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java

```diff
@@ -121,6 +121,11 @@ public OAuth2AccessToken readAccessToken(String accessToken) {
 	}
 
 	private String getAuthorizationHeader(String clientId, String clientSecret) {
+
+		if(clientId == null || clientSecret == null) {
+			logger.debug("Null Client ID or Client Secret detected. Endpoint that requires authentication will reject request with 401 error.");
```

Comment:
DEBUG is a bit of a weak signal, don't you think, given that this is almost certainly going to fail?